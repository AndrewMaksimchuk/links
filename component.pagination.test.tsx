import { describe, test, expect, mock } from 'bun:test'
import { renderToReadableStream } from 'hono/jsx/streaming'
import { createContext } from 'hono/jsx'
import { link } from './component.links.test'
import { Pagination } from './component.pagination'


mock.module('./page.dashboard', () => {
    return {
        LinksContext: createContext([]),
    };
})


describe('Component Pagination', () => {
    test('without links', async () => {
        const stream = renderToReadableStream(<Pagination activePage={1} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })


    test('with links', async () => {
        mock.module('./page.dashboard', () => {
            return {
                LinksContext: createContext(new Array(35).fill(link)),
            };
        })
        const stream = renderToReadableStream(<Pagination activePage={1} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })
})
