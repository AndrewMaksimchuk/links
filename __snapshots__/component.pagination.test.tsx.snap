// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Pagination without links 1`] = `""`;

exports[`Component Pagination with links 1`] = `"<div id="paginationSectionButtons" style="display:flex;flex-wrap:wrap;gap:1rem;margin-bottom:var(--pico-typography-spacing-vertical)" hx-swap-oob="outerHTML"><button style="--pico-form-element-spacing-vertical:0.125rem;--pico-form-element-spacing-horizontal:0.5rem" name="page" value="1" hx-post="/pagination-view-update" hx-push-url="/dashboard?page=1" hx-target="main" hx-swap="innerHTML">1</button><button style="--pico-form-element-spacing-vertical:0.125rem;--pico-form-element-spacing-horizontal:0.5rem" name="page" value="2" hx-post="/pagination-view-update" hx-push-url="/dashboard?page=2" hx-target="main" hx-swap="innerHTML">2</button></div>"`;
